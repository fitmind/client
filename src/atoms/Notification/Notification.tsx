import React, { useState } from 'react';
import { connect } from 'react-redux';
import { ApplicationState } from '../../redux/reducers/root.reducer';
import { Alert } from 'react-bootstrap';
import styled from 'styled-components';
import { NotificationInterface } from '../../interfaces/Notification.interface';

interface PropsFromState {
    notification: NotificationInterface;
}

const PositiveAlertDismissible: React.FC<NotificationInterface> = ({ header, body }) => {
    const [show, setShow] = useState(true);
    if (show) {
        return (
            <Alert variant="success" onClose={() => setShow(false)} dismissible>
                {header && <Alert.Heading>{header}</Alert.Heading>}
                {body && <p>{body}</p>}
            </Alert>
        );
    }
};

const NegativeAlertDismissible: React.FC<NotificationInterface> = ({ header, body }) => {
    const [show, setShow] = useState(true);
    if (show) {
        return (
            <Alert variant="danger" onClose={() => setShow(false)} dismissible>
                {header && <Alert.Heading>{header}</Alert.Heading>}
                {body && <p>{body}</p>}
            </Alert>
        );
    }
};

const Notification: React.FC<PropsFromState> = ({ notification }) => (
    <Wrapper>
        {notification.type === 'positive' && <PositiveAlertDismissible {...notification} />}
        {notification.type === 'error' && <NegativeAlertDismissible {...notification} />}
    </Wrapper>
);

const mapStateToProps = (state: ApplicationState) => ({
    notification: state.ui.notification,
});

export default connect(
    mapStateToProps,
    null,
)(Notification);

const Wrapper = styled.div`
    text-align: center;
`;
