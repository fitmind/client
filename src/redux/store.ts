import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
