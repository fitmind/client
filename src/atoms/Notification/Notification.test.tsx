import React from 'react';
import { shallow } from 'enzyme';
import store, { configureStore } from '../../redux/store';
import { Provider } from 'react-redux';
import Notification from './Notification';
import { storeInitialState } from '../../redux/reducers/root.reducer';
import { mergeDeepRight } from 'ramda';
import renderer from 'react-test-renderer';

describe('Loading Component', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Notification />
            </Provider>,
        );
    });
    it('should match the snapshot when the notification is positive', () => {
        const mergedProps = mergeDeepRight(storeInitialState, {
            ui: { notification: { type: 'positive', header: `Welcome Diego`, body: 'some body in the text' } },
        });
        const storeWithNotificationPositive = configureStore(mergedProps);
        const component = renderer.create(
            <Provider store={storeWithNotificationPositive}>
                <Notification />
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should match the snapshot when the notification is error', () => {
        const mergedProps = mergeDeepRight(storeInitialState, {
            ui: { notification: { type: 'error', header: `Welcome Diego`, body: 'some body in the text' } },
        });
        const storeWithNotificationNegative = configureStore(mergedProps);
        const component = renderer.create(
            <Provider store={storeWithNotificationNegative}>
                <Notification />
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should match the snapshot when the notification is blank', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Notification />
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
