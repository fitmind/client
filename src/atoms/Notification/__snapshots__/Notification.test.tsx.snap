// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should match the snapshot when the notification is blank 1`] = `
<div
  className="sc-bdVaJa kmXoBq"
/>
`;

exports[`Loading Component should match the snapshot when the notification is error 1`] = `
<div
  className="sc-bdVaJa kmXoBq"
/>
`;

exports[`Loading Component should match the snapshot when the notification is positive 1`] = `
<div
  className="sc-bdVaJa kmXoBq"
>
  <div
    className="fade alert alert-success alert-dismissible show"
    role="alert"
  >
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
      <span
        className="sr-only"
      >
        Close alert
      </span>
    </button>
    <div
      className="alert-heading h4"
    >
      Welcome Diego
    </div>
    <p>
      some body in the text
    </p>
  </div>
</div>
`;
